src/main/xsl/preprocess/conrefImpl.xsl

```diff
@@ -462,22 +462,25 @@ See the accompanying LICENSE file for applicable license.
                 <xsl:otherwise>
                   <xsl:copy>
                     <xsl:for-each select="@*">
-                      <xsl:if test="not(local-name(.) = 'id')">
-                        <xsl:choose>
-                          <xsl:when test="name() = 'href'">
-                            <!--@href need to update, not implement currently. @href may point to local part, but if @href pull into other file,
-                          then @href couldn't work correctly. This is the reason why @href need to update. We leave it as the future work.-->
-                            <xsl:apply-templates select=".">
-                              <xsl:with-param name="current-relative-path" select="$current-relative-path"/>
-                              <xsl:with-param name="topicid" select="$topicid"/>
-                              <xsl:with-param name="elemid" select="$elemid"/>
-                            </xsl:apply-templates>
-                          </xsl:when>
-                          <xsl:otherwise>
-                            <xsl:copy/>
-                          </xsl:otherwise>
-                        </xsl:choose>
-                      </xsl:if>
+                      <xsl:choose>
+                        <xsl:when test="name() = 'id'">
+                          <xsl:if test="contains(../@class,' topic/topic ')">
+                            <xsl:attribute name="id" select="generate-id(.)"/>
```

Comment:
This generates the new `@id` from `generate-id(@id)`. Should it instead be from the element, so `generate-id(..)` in this particular case? My preference is to use the element for source of identity, not attributes.

Reply:
Yep, not intentional, and more obvious / easily fixed with the refactoring suggested above.